router/http_test.go

```diff
@@ -410,61 +409,77 @@ func (s *S) TestStickyHTTPRoute(c *C) {
 }
 
 func (s *S) TestStickyHTTPRouteWebsocket(c *C) {
-	wshandler := func(w http.ResponseWriter, req *http.Request) {
-		w.Write([]byte("ok " + strings.TrimPrefix(req.URL.Path, "/")))
-	}
-	srv1 := httptest.NewServer(http.HandlerFunc(wshandler))
-	srv2 := httptest.NewServer(http.HandlerFunc(wshandler))
+	srv1 := httptest.NewServer(httpTestHandler("1"))
+	srv2 := httptest.NewServer(httpTestHandler("2"))
 	defer srv1.Close()
 	defer srv2.Close()
 
 	l, discoverd := newHTTPListener(c)
+	url := "http://" + l.Addr
 	defer l.Close()
+	defer discoverd.UnregisterAll()
 
 	addStickyHTTPRoute(c, l)
 
-	discoverdRegisterHTTP(c, l, srv1.Listener.Addr().String())
-	defer discoverd.UnregisterAll()
+	steps := []struct {
+		do          func()
+		backend     string
+		noSetCookie bool
```

Comment:
I think inverting this and renaming to `shouldSetCookie` or `setCookie` is less confusing.